### ISSUE TITLE
Running super slow on 4 a100 gpus

### ISSUE THREAD

**Issue by awrd2019** [mallorbc/Finetune_LLMs#16]
I'm finetuning the 7b llama model on your quotes dataset on a runpod instance with 4 a100 gpus yet its training at a rate of about 230 seconds per step. Any idea what could be going wrong here? Would be super grateful for your help. Also pasted below what I ran in command prompt as well as a screenshot of the w&b charts if that helps.

deepspeed --num_gpus=4 run_clm.py --deepspeed ds_config_stage3.json --model_name_or_path decapoda-research/llama-7b-hf --
train_file train.csv --validation_file validation.csv --do_train --do_eval --bf16 --overwrite_cache --evaluation_strategy=steps --output_dir finetuned --num_train_epochs 1
 --eval_steps 20 --gradient_accumulation_steps 32 --per_device_train_batch_size 1 --use_fast_tokenizer False --learning_rate 5e-06 --warmup_steps 10 --save_total_limit 1 --
save_steps 20 --save_strategy steps --load_best_model_at_end=True --block_size=200 --report_to=wandb

<img width="892" alt="Screenshot 2023-04-05 145739" src="https://user-images.githubusercontent.com/47894192/230178500-fc024b40-c80c-440c-be7f-4f4c918323f5.png">

**Comment by mallorbc** [comment]
Only use gradient_accumulation_steps if num_gpus*batch size is not what you want.  Raise the batch size and lower the gradient accumulation.  If the A100s are 80GB, also use stage 1 instead.

**Comment by awrd2019** [comment]
Will try this, thank you so much
